From 3fb8c85f2221b47e4918bcd3b2fbe0f7f5c03f19 Mon Sep 17 00:00:00 2001
From: Matthijs Van Assche <26224-Matthijs@users.noreply.drupalcode.org>
Date: Tue, 5 Mar 2024 10:03:32 +0000
Subject: [PATCH 1/2] Issue #3425688: Change file_validate functions to
 constraints

---
 src/Element/FileUploader.php | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/src/Element/FileUploader.php b/src/Element/FileUploader.php
index 340e1bd..a29a495 100644
--- a/src/Element/FileUploader.php
+++ b/src/Element/FileUploader.php
@@ -90,6 +90,11 @@ class FileUploader extends ManagedFile {
     ];
 
     // Prepare the attachment settings.
+    // The file_validate_* functions are for backwards compatibility they were deprecated in Drupal 10.2
+    $upload_validators = $element['#upload_validators'];
+    $extensions = $upload_validators['FileExtension']['extensions'] ?? ($upload_validators['file_validate_extensions'][0] ?? '');
+    $file_size = $upload_validators['FileSizeLimit']['fileLimit'] ?? ($upload_validators['file_validate_size'][0] ?? 0);
+
     $settings = [
       'provider' => $element['#upload_provider'],
       'name' => $element['#name'],
@@ -97,8 +102,8 @@ class FileUploader extends ManagedFile {
         'xhr' => $url->toString(),
         'validators' => [
           'limit' => $element['#cardinality'],
-          'extensions' => preg_filter('/^/', '.', explode(' ', $element['#upload_validators']['file_validate_extensions'][0] ?? NULL)),
-          'filesize' => $element['#upload_validators']['file_validate_size'][0] ?? 0,
+          'extensions' => preg_filter('/^/', '.', explode(' ', $extensions)),
+          'filesize' => file_size,
         ],
       ],
       'values' => $values,
-- 
GitLab


From e7aa96d7e35ce051c11099d0ee450886ba720f6c Mon Sep 17 00:00:00 2001
From: Matthijs Van Assche <26224-Matthijs@users.noreply.drupalcode.org>
Date: Tue, 5 Mar 2024 10:07:49 +0000
Subject: [PATCH 2/2] Issue #3425688: Fix missing $ sign

---
 src/Element/FileUploader.php | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/Element/FileUploader.php b/src/Element/FileUploader.php
index a29a495..0bf719c 100644
--- a/src/Element/FileUploader.php
+++ b/src/Element/FileUploader.php
@@ -90,7 +90,7 @@ class FileUploader extends ManagedFile {
     ];
 
     // Prepare the attachment settings.
-    // The file_validate_* functions are for backwards compatibility they were deprecated in Drupal 10.2
+    // Note: the file_validate_* functions are for backwards compatibility, they were deprecated in Drupal 10.2
     $upload_validators = $element['#upload_validators'];
     $extensions = $upload_validators['FileExtension']['extensions'] ?? ($upload_validators['file_validate_extensions'][0] ?? '');
     $file_size = $upload_validators['FileSizeLimit']['fileLimit'] ?? ($upload_validators['file_validate_size'][0] ?? 0);
@@ -103,7 +103,7 @@ class FileUploader extends ManagedFile {
         'validators' => [
           'limit' => $element['#cardinality'],
           'extensions' => preg_filter('/^/', '.', explode(' ', $extensions)),
-          'filesize' => file_size,
+          'filesize' => $file_size,
         ],
       ],
       'values' => $values,
-- 
GitLab

