diff --git a/src/Plugin/views/field/EntityFormField.php b/src/Plugin/views/field/EntityFormField.php
index 2381a41..a44f3b0 100644
--- a/src/Plugin/views/field/EntityFormField.php
+++ b/src/Plugin/views/field/EntityFormField.php
@@ -6,14 +6,17 @@ use Drupal\Component\Plugin\DependentPluginInterface;
 use Drupal\Component\Plugin\Factory\DefaultFactory;
 use Drupal\Core\Cache\Cache;
 use Drupal\Core\Cache\CacheableDependencyInterface;
+use Drupal\Core\Datetime\DrupalDateTime;
 use Drupal\Core\Entity\EntityFieldManagerInterface;
 use Drupal\Core\Entity\EntityInterface;
 use Drupal\Core\Entity\EntityTypeManagerInterface;
 use Drupal\Core\Field\FieldDefinitionInterface;
 use Drupal\Core\Field\WidgetPluginManager;
+use Drupal\Core\Form\FormBuilderInterface;
 use Drupal\Core\Form\FormStateInterface;
 use Drupal\Core\Language\LanguageManagerInterface;
 use Drupal\Core\Plugin\PluginDependencyTrait;
+use Drupal\Core\Url;
 use Drupal\views\Entity\Render\EntityTranslationRenderTrait;
 use Drupal\views\Plugin\DependentWithRemovalPluginInterface;
 use Drupal\views\Plugin\views\field\FieldPluginBase;
@@ -478,6 +481,7 @@ class EntityFormField extends FieldPluginBase implements CacheableDependencyInte
 
     // Only add the buttons if there are results.
     if (!empty($this->getView()->result)) {
+      $ajax_enabled = $this->getView()->display_handler->ajaxEnabled();
       $form[$this->options['id']]['#tree'] = TRUE;
       $form[$this->options['id']]['#entity_form_field'] = TRUE;
       foreach ($this->getView()->result as $row_index => $row) {
@@ -505,12 +509,97 @@ class EntityFormField extends FieldPluginBase implements CacheableDependencyInte
 
             // Hide field widget title.
             if ($this->options['plugin']['hide_title']) {
-              $form[$this->options['id']][$row_index][$field_name]['#attributes']['class'][] = 'views-entity-form-field-field-label-hidden';
+              if ($form[$this->options['id']][$row_index][$field_name]['#type'] == 'container') {
+                if (isset($form[$this->options['id']][$row_index][$field_name]['widget'][0]['container']['value']['#title'])) {
+                  unset($form[$this->options['id']][$row_index][$field_name]['widget'][0]['container']['value']['#title']);
+                }
+              }
+              else {
+                $form[$this->options['id']][$row_index][$field_name]['#attributes']['class'][] = 'views-entity-form-field-field-label-hidden';
+              }
             }
 
             // Hide field widget description.
             if ($this->options['plugin']['hide_description']) {
-              $form[$this->options['id']][$row_index][$field_name]['#attributes']['class'][] = 'views-entity-form-field-field-description-hidden';
+              if ($form[$this->options['id']][$row_index][$field_name]['#type'] == 'container') {
+                if (isset($form[$this->options['id']][$row_index][$field_name]['widget'][0]['container']['value']['#description'])) {
+                  unset($form[$this->options['id']][$row_index][$field_name]['widget'][0]['container']['value']['#description']);
+                }
+              }
+              else {
+                $form[$this->options['id']][$row_index][$field_name]['#attributes']['class'][] = 'views-entity-form-field-field-description-hidden';
+              }
+            }
+
+            // Set AJAX callback only if the view uses ajax.
+            if ($ajax_enabled === TRUE) {
+              // Set callback only on visible form elements.
+              if ($this->options['exclude'] == FALSE) {
+                $target_element = $form[$this->options['id']][$row_index][$field_name];
+                $path = \Drupal::service('path.current')->getPath();
+
+                $ajax_settings = [
+                  'callback' => [
+                    'Drupal\views_entity_form_field\Plugin\views\field\EntityFormField',
+                    'updateEntity',
+                  ],
+                  'event' => 'change',
+                  'disable-refocus' => TRUE,
+                  'progress' => [
+                    'type' => 'throbber',
+                    'message' => $this->t('Saving'),
+                  ],
+                  'url' => Url::fromUserInput($path),
+                  'options' => [
+                    'query' => [
+                      FormBuilderInterface::AJAX_FORM_REQUEST => TRUE,
+                    ],
+                  ],
+                ];
+                $ajax_settings['options']['query'] += \Drupal::request()->query->all();
+
+                // For a boolean (single checkbox).
+                if (isset($target_element['widget']['value'])) {
+                  $form[$this->options['id']][$row_index][$field_name]['widget']['value']['#ajax'] = $ajax_settings;
+                  $form[$this->options['id']][$row_index][$field_name]['widget']['value']['#relationship'] = $this->options['relationship'];
+                }
+                // For boolean (checkboxes, radios, toggle), number, text, date.
+                elseif (isset($target_element['widget'][0]['value'])) {
+                  $form[$this->options['id']][$row_index][$field_name]['widget'][0]['value']['#ajax'] = $ajax_settings;
+                  $form[$this->options['id']][$row_index][$field_name]['widget'][0]['value']['#relationship'] = $this->options['relationship'];
+                  // date triggeringElements lose the relationship
+                  $form[$this->options['id']][$row_index][$field_name]['widget'][0]['value']['#ajax']['#relationship'] = $this->options['relationship'];
+                }
+                // For end date (date) field.
+                elseif (isset($target_element['widget'][0]['container']['value'])) {
+                  $form[$this->options['id']][$row_index][$field_name]['widget'][0]['container']['value']['#ajax'] = $ajax_settings;
+                  $form[$this->options['id']][$row_index][$field_name]['widget'][0]['container']['value']['#relationship'] = $this->options['relationship'];
+                }
+                // For entity_reference (autocomplete).
+                elseif (isset($target_element['widget'][0]['target_id'])) {
+                  $form[$this->options['id']][$row_index][$field_name]['widget'][0]['target_id']['#ajax'] = $ajax_settings;
+                  $form[$this->options['id']][$row_index][$field_name]['widget'][0]['target_id']['#relationship'] = $this->options['relationship'];
+                }
+                // For entity_reference (tag style).
+                elseif (isset($target_element['widget']['target_id'])) {
+                  $form[$this->options['id']][$row_index][$field_name]['widget']['target_id']['#ajax'] = $ajax_settings;
+                  $form[$this->options['id']][$row_index][$field_name]['widget']['target_id']['#relationship'] = $this->options['relationship'];
+                }
+                // For entity_reference (select, checkboxes).
+                elseif (isset($target_element['widget']['#type'])) {
+                  $form[$this->options['id']][$row_index][$field_name]['widget']['#ajax'] = $ajax_settings;
+                  $form[$this->options['id']][$row_index][$field_name]['widget']['#relationship'] = $this->options['relationship'];
+                }
+                // And for the rest yet unknown.
+                else {
+                  $form[$this->options['id']][$row_index][$field_name]['widget']['#ajax'] = $ajax_settings;
+                  $form[$this->options['id']][$row_index][$field_name]['widget']['#relationship'] = $this->options['relationship'];
+                }
+                // Remove the save button from the form.
+                if (isset($form['actions']['submit'])) {
+                  $form['actions']['submit']['#access'] = FALSE;
+                }
+              }
             }
           }
         }
@@ -613,7 +702,8 @@ class EntityFormField extends FieldPluginBase implements CacheableDependencyInte
               $storage->save($entity);
               $rows_saved[$row_index] = $entity->label();
             }
-          } catch (\Exception $exception) {
+          }
+          catch (\Exception $exception) {
             $rows_failed[$row_index] = $entity->label();
           }
         }
@@ -646,9 +736,9 @@ class EntityFormField extends FieldPluginBase implements CacheableDependencyInte
   /**
    * Determines if an entity should be saved.
    *
-   * @param EntityInterface $entity
+   * @param \Drupal\Core\Entity\EntityInterface $entity
    *   The possibly modified entity in question.
-   * @param EntityInterface $original_entity
+   * @param \Drupal\Core\Entity\EntityInterface $original_entity
    *   The original unmodified entity.
    *
    * @return bool
@@ -674,4 +764,94 @@ class EntityFormField extends FieldPluginBase implements CacheableDependencyInte
     // Do nothing.
   }
 
+  /**
+   * AJAX callback to save te current entity.
+   *
+   * @param array $form
+   *   An associative array containing the structure of the form.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The current state of the form.
+   */
+  public static function updateEntity(array &$form, FormStateInterface $form_state) {
+    $trigger = $form_state->getTriggeringElement();
+
+    // The field, current row and value type are set in the #parents array.
+    $field = $trigger['#parents'][2];
+    $row = $trigger['#parents'][1];
+    $value_type = array_key_exists(4, $trigger['#parents']) ? $trigger['#parents'][4] : '';
+
+    // Get the proper value from the element.
+    if (isset($value_type) && $value_type == 'target_id') {
+      if (isset($trigger['#default_value']) && is_object($trigger['#default_value'])) {
+        $value = $trigger['#default_value']->id();
+      }
+      else {
+        // Need an empty array as value to delete the entity reference.
+        $value = [];
+      }
+    }
+    else {
+      if (isset($trigger['#type']) && $trigger['#type'] == 'date') {
+        $value_type = 'date';
+        $input_value = $form_state->getUserInput()['form_field_' . $field][$row][$field];
+        // For End date widgets the value is in a container.
+        if (isset($input_value[0]['container']['value'])) {
+          $date_value = $input_value[0]['container']['value'];
+        }
+        else {
+          $date_value = $input_value[0]['value'];
+        }
+
+        // If both values are empty - user has deleted the field values.
+        if (empty($date_value['date']) && empty($date_value['time'])) {
+          $value = '';
+        }
+        else {
+          $connector = !empty($date_value['time']) ? 'T' : '';
+          $value = $date_value['date'];
+          $format = 'Y-m-d';
+          if (isset($date_value['time'])) {
+            $value = $value . $connector . $date_value['time'];
+            $format = 'Y-m-d\TH:i:s';
+            // Get the Drupal site default timezone.
+            $timezone = new \DateTimeZone(date_default_timezone_get());
+            $value = new DrupalDateTime($value, $timezone);
+            // Convert to UTC and the DB storage format.
+            $value = $value->setTimeZone(new \DateTimeZone('UTC'))->format($format);
+          }
+        }
+      }
+      else {
+        $value = $trigger['#value'];
+      }
+    }
+
+    // Get the entity to update from the views row.
+    if (!empty($trigger['#relationship']) && $trigger['#relationship'] != 'none') {
+      $entity = $form['output'][0]['#rows'][$row]->_relationship_entities[$trigger['#relationship']];
+    }
+    elseif (!empty($trigger['#ajax']['#relationship'])) {
+      $entity = $form['output'][0]['#rows'][$row]->_relationship_entities[$trigger['#ajax']['#relationship']];
+    }
+    else {
+      $entity = $form['output'][0]['#rows'][$row]->_entity;
+    }
+
+    // reload the entity
+    $entity_type = $entity->getEntityTypeId();
+    $entity = \Drupal::entityTypeManager()->getStorage($entity_type)->load($entity->id());
+
+    // If the entity has the field - write the value.
+    if (is_object($entity) && $entity->hasField($field)) {
+      if (in_array($value_type, ['target_id', 'date'])) {
+        $entity->set($field, [$value_type = $value]);
+      }
+      else {
+        $entity->set($field, $value);
+      }
+      $entity->save();
+      return ['#markup' => "Ok"];
+    }
+  }
+
 }
